From: Kees Cook <kees@ubuntu.com>
Date: Fri, 5 Jun 2026 17:15:24 +0800
Subject: pro-actively disable LD_AUDIT for setuid binaries, regardless

of where the libraries are loaded from. This is to try to make sure that
CVE-2010-3856 cannot sneak back in. Upstream is unlikely to take this,
since it limits the functionality of LD_AUDIT.
---
 elf/rtld.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/elf/rtld.c b/elf/rtld.c
index 29e7a4d..fa9cc8d 100644
--- a/elf/rtld.c
+++ b/elf/rtld.c
@@ -264,7 +264,7 @@ audit_list_next (struct audit_list *list)
 	++list->current_tail;
 
       /* If the name is valid, return it.  */
-      if (dso_name_valid_for_suid (list->fname))
+      if (! __glibc_unlikely (__libc_enable_secure) && dso_name_valid_for_suid (list->fname))
 	return list->fname;
 
       /* Otherwise wrap around to find the next list element. .  */
